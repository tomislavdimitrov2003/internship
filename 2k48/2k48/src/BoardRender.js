import React from 'react';


export function RenderBoard(board, score, message) {
    return (
        <div>
            <div className="score">Score: {score}</div>
            <table border="1px solid black" border-collapse="collapse">
                <tbody>
                    {board.map((row, i) => (<Row key={i} row={row} />))}
                </tbody>
            </table>
            <div>{message}</div>
        </div>
    );
}

const Row = ({ row }) => {
    return (
        <tr>
            {row.map((cell, i) => (<Cell key={i} cellValue={cell} />))}
        </tr>
    );
};

const Cell = ({ cellValue }) => {
    let value = (cellValue === 0) ? '' : cellValue;

    return (
        <td
            border="1px solid black"
            border-collapse="collapse"
            width="50"
            height="50"
            align="center"
            style={{ backgroundColor: color(cellValue) }}>
            <div>{value}</div>
        </td>
    );
};

var color = function (value) {
    if (!value) {
        return "white";
    }
    var blue = ((Math.log2(value) + 1) % 2) * 128;
    var green = (Math.log2(value) * 40 + 100) % 256;
    var red = (Math.log2(value) % 2) * 128;

    return "rgb(" + red + ", " + green + ", " + blue + ")";
}