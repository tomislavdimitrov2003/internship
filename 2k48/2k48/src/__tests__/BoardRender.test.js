import { renderBoard } from "../BoardRender";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import React from 'react';


it('reanders board correctly', () => {
    const board = [[0, 0, 0, 0],
                   [0, 0, 0, 0],
                   [0, 0, 0, 0],
                   [0, 0, 0, 0]];
    const score = 0;
    const message = "Test Message";               
    const wrapper = shallow(<renderBoard board={board} score={score} message={message}/>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});