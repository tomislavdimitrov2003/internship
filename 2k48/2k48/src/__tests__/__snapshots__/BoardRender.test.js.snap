// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reanders board correctly 1`] = `
<renderBoard
  board={
    Array [
      Array [
        0,
        0,
        0,
        0,
      ],
      Array [
        0,
        0,
        0,
        0,
      ],
      Array [
        0,
        0,
        0,
        0,
      ],
      Array [
        0,
        0,
        0,
        0,
      ],
    ]
  }
  message="Test Message"
  score={0}
/>
`;
